File: datasets/php/HumanEval_118_get_closest_vowel.php
<?php
// You are given a word. Your task is to find the closest vowel that stands between 
// two consonants from the right side of the word (case sensitive).
// Vowels in the beginning and ending doesn't count. Return empty string if you didn't
// find any vowel met the above condition. 
// You may assume that the given string contains English letter only.
// Example:
// get_closest_vowel("yogurt") ==> "u"
// get_closest_vowel("FULL") ==> "U"
// get_closest_vowel("quick") ==> ""
// get_closest_vowel("ab") ==> ""
function get_closest_vowel($word) {
    $vowels = ['a', 'e', 'i', 'o', 'u'];
    $consonants = ['b', 'c', 'd', 'f', 'g', 'h', 'j', 'k', 'l', 'm', 'n', 'p', 'q', 'r', 's', 't', 'v', 'w', 'x', 'y', 'z'];
    $word = strtolower($word);
    $word = str_split($word);
    $word = array_reverse($word);
    $vowel_index = 0;
    $consonant_index = 0;
    $vowel_found = false;
    $consonant_found = false;
    $vowel_position = 0;
    $consonant_position = 0;
    $vowel_distance = 0;
    $consonant_distance = 0;
    $vowel_closest = '';
    $consonant_closest = '';
    $vowel_closest_distance = 0;
    $consonant_closest_distance = 0;
    $vowel_closest_position = 0;
    $consonant_closest_position = 0;
    $vowel_closest_index = 0;
    $consonant_closest_index = 0;
    $vowel_closest_found = false;
    $consonant_closest_found = false;
    $vowel_closest_distance = 0;
    $consonant_closest_distance = 0;
    $vowel_closest_position = 0;
    $consonant_closest_position = 0;
    $vowel_closest_index = 0;
    $consonant_closest_index = 0;
    $vowel_closest_found = false;
    $consonant_closest_found = false;
    for ($i = 0; $i < count($word); $i++)

function candidate(...$args) {
    return get_closest_vowel(...$args);
}

function test(): void {
    if (candidate('yogurt') !== 'u') { throw new Exception("Test failed!"); }
    if (candidate('full') !== 'u') { throw new Exception("Test failed!"); }
    if (candidate('easy') !== '') { throw new Exception("Test failed!"); }
    if (candidate('eAsy') !== '') { throw new Exception("Test failed!"); }
    if (candidate('ali') !== '') { throw new Exception("Test failed!"); }
    if (candidate('bad') !== 'a') { throw new Exception("Test failed!"); }
    if (candidate('most') !== 'o') { throw new Exception("Test failed!"); }
    if (candidate('ab') !== '') { throw new Exception("Test failed!"); }
    if (candidate('ba') !== '') { throw new Exception("Test failed!"); }
    if (candidate('quick') !== '') { throw new Exception("Test failed!"); }
    if (candidate('anime') !== 'i') { throw new Exception("Test failed!"); }
    if (candidate('Asia') !== '') { throw new Exception("Test failed!"); }
    if (candidate('Above') !== 'o') { throw new Exception("Test failed!"); }
}

test();

File: datasets/java/HumanEval_118_get_closest_vowel.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
class Problem {
private static <T> T getDefaultValue(Class<T> clazz) {
return (T) Array.get(Array.newInstance(clazz, 1), 0);
}    // You are given a word. Your task is to find the closest vowel that stands between 
    // two consonants from the right side of the word (case sensitive).
    // Vowels in the beginning and ending doesn't count. Return empty string if you didn't
    // find any vowel met the above condition. 
    // You may assume that the given string contains English letter only.
    // Example:
    // get_closest_vowel("yogurt") ==> "u"
    // get_closest_vowel("FULL") ==> "U"
    // get_closest_vowel("quick") ==> ""
    // get_closest_vowel("ab") ==> ""
    public static String get_closest_vowel(String word) {
        String vowels = "aeiou";
        int i = word.length() - 1;
        while (i >= 0 && vowels.indexOf(word.charAt(i)) == -1) {
            i--;
        }
        if (i < 0) {
            return "";
        }
        int j = i - 1;
        while (j >= 0 && vowels.indexOf(word.charAt(j)) == -1) {
            j--;
        }
        if (j < 0) {
            return word.charAt(i) + "";
        }
        return word.charAt(j) + word.charAt(i);
    }
    // You are given a string. Your task is to count the number of words in the string.
    // A word is a sequence of non-whitespace characters.
    // Example:
    // count_words("Hello World!") ==> 2
    // count_words("The quick brown fox jumps over the lazy dog.") ==> 8
    public static int count_words(String str) {
        int count = 0;
        for (int i = 0; i < str.length(); i++) {
            if (Character.isWhitespace(str.charAt(i))) {
                count++;
            }
        }
        return count + 1;
    }
    // You are given a string. Your task is to count the number of words in the string.
    // A word is a sequence of non-whitespace characters.
    // Example:
    // count_words("Hello World!") ==> 2
    // count_words("The quick brown fox jumps over the lazy dog.") ==> 8
    public static int count_words(String str) {
        int count = 0;
        for (int i = 0; i < str.length(); i++) {
            if (Character.isWhitespace(str.charAt(i))) {
                count++;
            }
        }
        return count + 1;
    }
    // You are given a string. Your task is to count the number of words in the string.
    // A word is a

    }
    public static void main(String[] args) {
    get_closest_vowel(("yogurt")).equals(("u"));
    get_closest_vowel(("full")).equals(("u"));
    get_closest_vowel(("easy")).equals((""));
    get_closest_vowel(("eAsy")).equals((""));
    get_closest_vowel(("ali")).equals((""));
    get_closest_vowel(("bad")).equals(("a"));
    get_closest_vowel(("most")).equals(("o"));
    get_closest_vowel(("ab")).equals((""));
    get_closest_vowel(("ba")).equals((""));
    get_closest_vowel(("quick")).equals((""));
    get_closest_vowel(("anime")).equals(("i"));
    get_closest_vowel(("Asia")).equals((""));
    get_closest_vowel(("Above")).equals(("o"));
    }

}
